### PR TITLE
Workaround crashes in MAME's zlib decompression code

### PR DESCRIPTION
**Pull request by CasualPokePlayer**
"Fixes" #3615

### CODE REVIEW

**Comment by vadosnaprimer** [comment]
Do we need to put it back into a zip? MAME can load from a folder if it's named correctly. If it's possible to send it such folders it would be cool.

**Comment by CasualPokePlayer** [comment]
If the zip contains mixed deflate and lzma compression, then this code would be needed (or bring in SharpCompress into the Emulation.Cores project), as ZipArchive doesn't support anything other than deflate.

File: src/BizHawk.Emulation.Cores/Arcades/MAME/MAME.cs [hunk 1]
```diff
@@ -129,15 +131,50 @@ private void StartMAME(List<IRomAsset> roms)
...
+							using var newEntryStream = newEntry.Open();
+							newEntryStream.Write(contents, 0, contents.Length);
+						}
+						catch
+						{
+							// ignored
+						}
+					}
+
+					zip.Dispose(); // this actually modifies the MemoryStream
```
Comment (nattthebear) [inline src/BizHawk.Emulation.Cores/Arcades/MAME/MAME.cs:162]: Style nit --  this would be better as a `using` block

Comment (CasualPokePlayer) [reply src/BizHawk.Emulation.Cores/Arcades/MAME/MAME.cs:162]: Dispose needs to be called before ret.ToArray() is done. In hindsight, I would suppose another pair of parens would work for that.

Comment (nattthebear) [reply src/BizHawk.Emulation.Cores/Arcades/MAME/MAME.cs:162]: This is a using block, as opposed to a using declaration:

```C#
using (var zip = new ZipArchive(ret, ZipArchiveMode.Update, leaveOpen: true))
{
  // ...
}
```

**Pull request merged**